lib/src/components/NxAccordion/stateful/__tests__/NxStatefulAccordion.test.tsx

```diff
@@ -4,46 +4,213 @@
-    expect(component).toHaveProp('className', 'foo');
-    expect(component).toHaveProp('id', 'bar');
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
   it('sets the NxAccordion open prop to the defaultOpen prop initially', function() {
-    expect(getShallowComponent()).toHaveProp('open', false);
-    expect(getShallowComponent({ defaultOpen: false })).toHaveProp('open', false);
-    expect(getShallowComponent({ defaultOpen: true })).toHaveProp('open', true);
+    expect(renderEl()).not.toHaveAttribute('open');
+    expect(renderEl({ defaultOpen: false })).not.toHaveAttribute('open');
+    expect(renderEl({ defaultOpen: true })).toHaveAttribute('open');
+  });
+
+  it('renders non-header children in content wrapper', function() {
+    const el = renderEl({
+      children: (
+        <>
+          <NxAccordion.Header>
+            <span>Foo</span>
+          </NxAccordion.Header>
+          <span className="bar">Bar</span>
+        </>
+      )
+    })!;
+
+    expect(el.querySelector('summary .bar')).not.toBeInTheDocument();
+    expect(el.querySelector('.bar')).toBeInTheDocument();
+  });
+
+  describe('Header', function() {
+    const renderEl = rtlRenderElement(NxAccordion.Header, {});
+
+    it('sets the provided className', function() {
+      const el = renderEl()!;
+      const customEl = renderEl({ className: 'foo' })!;
+
+      expect(customEl).toHaveClass('foo');
+
+      for (const cls of Array.from(el.classList)) {
+        expect(customEl).toHaveClass(cls);
+      }
+    });
+
+    it('renders the NxAccordion.Header as a <summary>', function() {
+      const el = renderEl({ id: 'foo' })!;
+      expect(el.tagName).toBe('SUMMARY');
+      expect(el).toHaveAttribute('id', 'foo');
+    });
+
+    it('sets aria-controls to the accordion id when id is not specified', function() {
+      const { container } = quickRender({
+        children: (
+          <NxAccordion.Header></NxAccordion.Header>
+        )
+      });
+      const id = container.querySelector('DETAILS')?.getAttribute('id');
+      expect(container.querySelector('SUMMARY')).toHaveAttribute('aria-controls', id);
+    });
+
+    it('sets aria-controls to the specified accordion id', function() {
+      const { container } = quickRender({
+        id: 'foo',
+        children: (
+          <NxAccordion.Header>
+            <span>Foo</span>
+          </NxAccordion.Header>
+        )
+      });
+
+      expect(container.querySelector('SUMMARY')).toHaveAttribute('aria-controls', 'foo');
+    });
   });
 
-  it('toggles the NxAccordion open prop when the NxAccordion onToggle callback is called', function() {
-    const component = getShallowComponent();
+  describe('onToggle', function() {
```

Comment:
The way these tests are organized is a little bit confusing.  I'd change this wrapping `describe` to be about "accordion header click" because the tests below test more than just `onToggle`, they test all of the behaviors that result from clicking the accordion.  Then, in the "when the accordion is currently closed" block, I'd separate the testing of `onToggle` and the state change into two different tests.  Then in the "when the accordion is currently open" block I'd follow the same pattern - basically make the tests identical except for the minimum changes to the setup and assertions.

Reply:
> The way these tests are organized is a little bit confusing. I'd change this wrapping `describe` to be about "accordion header click" because the tests below test more than just `onToggle`, they test all of the behaviors that result from clicking the accordion. Then, in the "when the accordion is currently closed" block, I'd separate the testing of `onToggle` and the state change into two different tests. Then in the "when the accordion is currently open" block I'd follow the same pattern - basically make the tests identical except for the minimum changes to the setup and assertions.

Like this?  7fe3a943195f7cb77776aec04c08b9a636b59083